### PR TITLE
Fix scrolling and swiping issues

### PR DESCRIPTION
**Pull request by marekrozmus**
Closes: #40

### CODE REVIEW

File: src/SwipeableListItem.js [hunk 1]
```diff
@@ -8,36 +8,56 @@ const SwipeActionPropType = PropTypes.shape({
...
-    this.left = 0;
-
-    this.fpsInterval = 1000 / 60;
     this.startTime = null;
+
+    this.resetState();
   }
 
+  resetState = () => {
+    this.dragStartPoint = { x: -1, y: -1 };
```
Comment (przemyslawzalewski) [inline src/SwipeableListItem.js]: Please use some neutral values like `0`.

File: src/SwipeableListItem.js [hunk 2]
```diff
@@ -136,43 +171,74 @@ class SwipeableListItem extends PureComponent {
...
-      event.stopPropagation();
-      const delta = event.clientX - this.dragStartX;
+  dragStartedWithinItem = () => {
+    const { x, y } = this.dragStartPoint;
 
-      if (this.shouldMoveItem(delta)) {
-        this.left = delta;
-      }
-    }
+    return x !== -1 && y !== -1;
```
Comment (przemyslawzalewski) [inline src/SwipeableListItem.js]: Please use `getBoundingClientRect` to compute proper hit-testing.

File: src/SwipeableListItem.js (hunk 1 above)
Comment (marekrozmus) [reply src/SwipeableListItem.js]: user can be lucky and click in point (0, 0) that is why these are negative numbers :)

File: src/SwipeableListItem.js (hunk 2 above)
Comment (marekrozmus) [reply src/SwipeableListItem.js]: x and y are set only when item is clicked and the event of clicking comes from event listener added on that item

**Pull request merged**